apps/src/templates/instructions/InstructionsCSF.jsx

```diff
@@ -388,138 +250,65 @@ class InstructionsCSF extends React.Component {
     return this.instructions.parentElement;
   };
 
-  /**
-   * Manually scroll instructions to bottom. When we have multiple
-   * elements in instructions, "bottom" is defined as 10 pixels above
-   * the top of the bottommost element. This is so we don't scroll past
-   * the beginning of a long element, and so we scroll to a position
-   * such that you can see that there is an element above it which you
-   * can scroll up to.
-   */
-  scrollInstructionsToBottom() {
-    const instructions = this.instructions;
-    const contentContainer = instructions.parentElement;
-    if (instructions.children.length > 1) {
-      const lastChild = instructions.children[instructions.children.length - 1];
-      scrollTo(contentContainer, lastChild.offsetTop - 10);
-    } else {
-      scrollBy(contentContainer, contentContainer.scrollHeight);
-    }
-  }
-
-  /**
-   * Handle a click to the hint display bubble (lightbulb)
-   */
-  handleClickBubble = () => {
-    // If we don't have authored hints to display, clicking bubble shouldnt do anything
-    if (this.props.hasAuthoredHints && this.props.hasUnseenHint) {
-      this.setState({
-        promptForHint: true
-      });
-      if (this.props.collapsed) {
-        this.props.handleClickCollapser();
-      }
-    }
-  };
-
   dismissHintPrompt = () => {
     this.setState({
       promptForHint: false
     });
   };
 
-  showHint = () => {
-    this.dismissHintPrompt();
-    this.props.showNextHint();
-    this.props.clearFeedback();
-  };
-
-  shouldDisplayHintPrompt() {
+  shouldDisplayHintPrompt = () => {
     return this.state && this.state.promptForHint && !this.props.collapsed;
-  }
-
-  shouldDisplayCollapserButton() {
-    // if we have "extra" (non-instruction) content, we should always
-    // give the option of collapsing it
-    if (
-      this.props.hints.length ||
-      this.shouldDisplayHintPrompt() ||
-      this.props.feedback
-    ) {
-      return true;
-    }
-
-    // Otherwise, only show the button if we have two versions of
-    // instruction we want to toggle between
-    return this.props.longInstructions && !this.shouldIgnoreShortInstructions();
-  }
+  };
 
-  shouldIgnoreShortInstructions() {
-    // If we have no short instructions, always ignore them.
-    if (!this.props.shortInstructions) {
-      return true;
-    }
+  hasShortInstructions = () => {
```

Comment:
the logic here is a bit confusing and would benefit from comments. some of the ones you removed seem pretty helpful, though they probably need to be updated.

separating out `shortAndLongInstructionsAreEquivalent` into its own function seems useful for giving a name to that part of the logic, but that usefulness is muddied by the checks for short/long instructions at the top of it (the first of which is redundant since you just checked `!!this.props.shortInstructions`, so i'd suggest keeping that part of the logic in this function and having `shortAndLongInstructionsAreEquivalent` only do the levenshtein check. however, since i still think that logic needs a comment despite the helpful function name, at that point i'm not sure it's worth breaking it out into its own function anymore, because the sheer number of helper functions in this component is already a lot to process.

tl;dr i think this would be clearer if you perform the checks for the presence of short/long instructions and return early if possible (with comments), then return the result of the levenshtein check if not (again, with comments)

Reply:
Agreed, this logic is very strange and confusing. I was hoping that if the helper functions were concise and clear that the comments wouldn't be necessary, but it seems like they'd still be valuable so I'll add some back in. I'll also try combining the `shortAndLongInstructionsAreEquivalent ` with `hasShortInstructions` again and see if that clarifies things.

Reply:
totally. at the very least, preserving the comment about "Levenshtein Edit Distance" would be helpful for those of us who wouldn't automatically know what `levenshtein()` is doing.

not sure if this would work for all of the use cases of this function, but i'm wondering if renaming this to `hasShortAndLongInstructions` and updating the logic to do that full calculation would make this clearer, rather than deferring that to where this is used in the child.